src/Form/Form.js

```diff
@@ -467,6 +499,15 @@ class Form {
         })
     }
 
+    execOnInput (fn, input, shouldCheckForDecorate) {
+        let canExecute = true
+        // sometimes we want to execute even if we didn't decorate
+        if (shouldCheckForDecorate) {
+            canExecute = isFieldDecorated(input)
+        }
+        if (canExecute) fn(input)
+    }
+
```

Comment:
I think you will have lost type safety here by extracting this. You can replicate the types if you want to keep this :)

Reply:
Ah, this was an overkill anyway:
https://github.com/duckduckgo/duckduckgo-autofill/pull/619/commits/3b8ee35c3f2f5461520cc101a8936c484088a7e0

reverted the change.